UnityProject/Assets/Scripts/Items/Weapons/SawnOff.cs

```diff
@@ -58,42 +58,40 @@ public bool WillInteract(InventoryApply interaction, NetworkSide side)
 
 	public void ServerPerformInteraction(InventoryApply interaction)
 	{
-		if (interaction.TargetObject == gameObject && interaction.IsFromHandSlot)
+		if (interaction.TargetObject != gameObject && !interaction.IsFromHandSlot) return;
+
+		//TODO: switch this trait to the circular saw when that is implemented
+		if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Welder) && gunComp.FireCountDown == 0)
```

Comment:
actually do we need to check the trait again? I saw it was being checked in WillInteract

Reply:
it is setup so willinteract handles clientside checks and serverperform handles serverside
it's done this way in most of the other gun code due to the server needing to do several different things depending on the objects used within the interaction, so it only checks it once clientside and once serverside instead of twice serverside.
kept it this way for consistency

Reply:
afaik WillInteract runs on both client first and server later but alright. Are you ready to merge now?

Reply:
should be good now, yeah